cpp/src/IceSSL/SChannelEngine.cpp

```diff
@@ -392,151 +396,15 @@ namespace
-        }
-        else if (name == "AES_192")
-        {
-            return CALG_AES_192;
-        }
-        else if (name == "AES_256")
-        {
-            return CALG_AES_256;
-        }
-        else if (name == "AGREEDKEY_ANY")
-        {
-            return CALG_AGREEDKEY_ANY;
-        }
-        else if (name == "CYLINK_MEK")
-        {
-            return CALG_CYLINK_MEK;
-        }
-        else if (name == "DES")
-        {
-            return CALG_DES;
-        }
-        else if (name == "DESX")
-        {
-            return CALG_DESX;
-        }
-        else if (name == "DH_EPHEM")
-        {
-            return CALG_DH_EPHEM;
-        }
-        else if (name == "DH_SF")
-        {
-            return CALG_DH_SF;
-        }
-        else if (name == "DSS_SIGN")
-        {
-            return CALG_DSS_SIGN;
-        }
-        else if (name == "ECDH")
-        {
-            return CALG_ECDH;
-        }
-        else if (name == "ECDH_EPHEM")
-        {
-            return ICESSL_CALG_ECDH_EPHEM;
-        }
-        else if (name == "ECDSA")
-        {
-            return CALG_ECDSA;
-        }
-        else if (name == "HASH_REPLACE_OWF")
-        {
-            return CALG_HASH_REPLACE_OWF;
-        }
-        else if (name == "HUGHES_MD5")
-        {
-            return CALG_HUGHES_MD5;
-        }
-        else if (name == "HMAC")
-        {
-            return CALG_HMAC;
-        }
-        else if (name == "MAC")
-        {
-            return CALG_MAC;
-        }
-        else if (name == "MD2")
-        {
-            return CALG_MD2;
-        }
-        else if (name == "MD4")
-        {
-            return CALG_MD4;
-        }
-        else if (name == "MD5")
-        {
-            return CALG_MD5;
-        }
-        else if (name == "NO_SIGN")
-        {
-            return CALG_NO_SIGN;
-        }
-        else if (name == "RC2")
-        {
-            return CALG_RC2;
-        }
-        else if (name == "RC4")
-        {
-            return CALG_RC4;
-        }
-        else if (name == "RC5")
-        {
-            return CALG_RC5;
-        }
-        else if (name == "RSA_KEYX")
-        {
-            return CALG_RSA_KEYX;
-        }
-        else if (name == "RSA_SIGN")
-        {
-            return CALG_RSA_SIGN;
-        }
-        else if (name == "SHA1")
-        {
-            return CALG_SHA1;
-        }
-        else if (name == "SHA_256")
-        {
-            return CALG_SHA_256;
-        }
-        else if (name == "SHA_384")
-        {
-            return CALG_SHA_384;
-        }
-        else if (name == "SHA_512")
-        {
-            return CALG_SHA_512;
-        }
-        return 0;
-    }
 }
 
 SChannel::SSLEngine::SSLEngine(const IceInternal::InstancePtr& instance)
     : IceSSL::SSLEngine(instance),
       _rootStore(0),
       _chainEngine(0),
-      _strongCrypto(false)
+      _strongCrypto(false),
+      _clientCredentials({0, 0}),
+      _serverCredentials({0, 0})
```

Comment:
Assuming these are pointers.
```suggestion
      _clientCredentials({nullptr, nullptr}),
      _serverCredentials({nullptr, nullptr})
```

Reply:
They are not.